tapdb/assets_store.go

```diff
@@ -1014,20 +1042,37 @@ func (a *AssetStore) QueryBalancesByAsset(ctx context.Context,
 // QueryAssetBalancesByGroup queries the asset balances for asset groups or
 // alternatively for a selected one that matches the passed filter.
 func (a *AssetStore) QueryAssetBalancesByGroup(ctx context.Context,
-	groupKey *btcec.PublicKey) (map[asset.SerializedKey]AssetGroupBalance,
+	groupKey *btcec.PublicKey,
+	includeLeased bool) (map[asset.SerializedKey]AssetGroupBalance,
 	error) {
 
-	var groupFilter []byte
+	// We'll now map the application level filtering to the type of
+	// filtering our database query understands.
+	assetBalancesFilter := QueryAssetBalancesByGroupFilters{
+		Now: sql.NullTime{
+			Time:  a.clock.Now().UTC(),
+			Valid: true,
+		},
+	}
+
+	// By default, we only show assets that are not leased.
+	if !includeLeased {
```

Comment:
Would be nice to also add some unit tests for the flag, especially around the three different values for the `leased` flag on the DB level (`NULL` = no filter, `true`  = only leased, `false` = only non-leased).

Reply:
Added a unit test. Although the state 'only leased' can't be hit, because we never run the query with `leased = true`. 
```
if !includeLeased {
	assetBalancesFilter.Leased = sqlBool(false)
}
```
All other states are extensively tested with grouped and leased assets mixed.